### PR TITLE
extension: Delete working directory on uninstall

### LINKED ISSUE #17126: uninstall extension but don't remove files in work dir
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

After deleting the extension, there are still some residual directories and files related to the extension in the `work` directory.
For example, I deleted the `dockerfile` extension：
image

### Environment

Apple M1 sonoma 14.5

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

### PR DESCRIPTION
**Pull request by bestgopher**
Closes #17126 

Release Notes:

- N/A

### CODE REVIEW

File: crates/extension/src/extension_store.rs [hunk 1]
```diff
@@ -772,6 +772,7 @@ impl ExtensionStore {
 
     pub fn uninstall_extension(&mut self, extension_id: Arc<str>, cx: &mut ModelContext<Self>) {
         let extension_dir = self.installed_dir.join(extension_id.as_ref());
+        let work_dir = self.wasm_host.clone().work_dir.join(extension_id.as_ref());
```
Comment (maxdeviant) [inline crates/extension/src/extension_store.rs:775]: We don't need to `clone` here:

```suggestion
        let work_dir = self.wasm_host.work_dir.join(extension_id.as_ref());
```

**Comment by bestgopher** [comment]
`clone` has been removed.

File: crates/extension/src/extension_store.rs (hunk 1 above)
Comment (bestgopher) [reply crates/extension/src/extension_store.rs]: done!

**Review by maxdeviant — APPROVED** [review]
Thanks!

**Pull request merged**